source/rust_verify/tests/atomic_lib.rs

```diff
@@ -0,0 +1,637 @@
+#![feature(rustc_private)]
+#[macro_use]
+mod common;
+use common::*;
+
+test_verify_one_file! {
+    #[test] test_atomic_u64 code! {
+        use crate::pervasive::{atomics::*};
+        use crate::pervasive::{modes::*};
+
+        pub fn foo() {
+            let (at, proof(mut perm)) = PAtomicU64::new(5);
+
+            assert(perm.value == 5);
+
+            let l = at.load(&perm);
+            assert(l == 5);
+
+            at.store(&mut perm, 6);
+            assert(perm.value == 6);
+
+            let l = at.swap(&mut perm, 9);
+            assert(l == 6);
+            assert(perm.value == 9);
+
+            let l = at.fetch_add_wrapping(&mut perm, 0xffff_ffff_ffff_ffff);
+            assert(l == 9);
+            assert(perm.value == 8);
+
+            let l = at.fetch_sub_wrapping(&mut perm, 0xffff_ffff_ffff_ffff);
+            assert(l == 8);
+            assert(perm.value == 9);
+
+            let l = at.fetch_or(&mut perm, 2);
+            assert(l == 9);
+            assert_bit_vector((9 as u64 | 2 as u64) == 11 as u64);
+            assert(perm.value == 11);
+
+            let l = at.fetch_and(&mut perm, 6);
+            assert(l == 11);
+            assert_bit_vector((11 as u64 & 6 as u64) == 2 as u64);
+            assert(perm.value == 2);
+
+            let l = at.fetch_xor(&mut perm, 3);
+            assert(l == 2);
+            assert_bit_vector((2 as u64 ^ 3 as u64) == 1 as u64);
+            assert(perm.value == 1);
+
+            let l = at.fetch_max(&mut perm, 5);
+            assert(l == 1);
+            assert(perm.value == 5);
+
+            let l = at.fetch_max(&mut perm, 4);
+            assert(l == 5);
+            assert(perm.value == 5);
+
+            let l = at.fetch_min(&mut perm, 4);
+            assert(l == 5);
+            assert(perm.value == 4);
+
+            let l = at.fetch_min(&mut perm, 7);
+            assert(l == 4);
+            assert(perm.value == 4);
+
+            let l = at.into_inner(perm);
+            assert(l == 4);
+
+            assert(false); // FAILS
```

Comment:
What is the purpose of these `assert(false)`s?

Unfortunately, due to the fact that verus currently doesn't have the same error reporting behavior as dafny (which tries its best to report the earliest assertion failure), this may hide an assertion elsewhere in the SMT query (for one of the other `assert`s). This is why we generally separate failing cases into separate functions, just to make sure they may not hide other failures in the postive tests.

Reply:
It's a "smoke test" to make sure there aren't contradictory post-conditions in the trusted methods. It's not a substitute for careful review ... but it's also cheap.

Reply:
anyway, given what you say, it sounds better to remove it

Reply:
> It's a "smoke test" to make sure there aren't contradictory post-conditions in the trusted methods. It's not a substitute for careful review ... but it's also cheap.

Ah, makes sense, thanks!
I think I maybe have a solution that addresses my concerns and also preserves the smoke test: https://github.com/secure-foundations/verus/pull/46
If that looks good, please merge it before merging this.